File: src/main/java/dev/rono/proxychat/utils/CooldownRunnable.java
package dev.rono.proxychat.utils;

import net.md_5.bungee.api.connection.ProxiedPlayer;

import java.util.UUID;
import java.util.concurrent.TimeUnit;

import static java.lang.Math.abs;

public class CooldownRunnable implements Runnable {

    public UUID player;
    public ToggleUtils utils;
    private final Long delay;

    public Long startTime = System.currentTimeMillis();

    public CooldownRunnable(ProxiedPlayer player, ToggleUtils utils, Long delay) {
        this.player = player.getUniqueId();
        this.utils = utils;
        this.delay = delay;
    }

    @Override
    public void run() {
        this.utils.toggleDelayOff(this.player);
    }

    public String getTime() {
        return String.format("%s", abs(TimeUnit.MILLISECONDS.toSeconds((System.currentTimeMillis() - this.startTime) - delay)));
    }
}

File: core/src/main/kotlin/dev/rono/proxychat/core/utils/CooldownRunnable.kt
package dev.rono.proxychat.core.utils

import java.util.*
import java.util.concurrent.TimeUnit
import kotlin.math.abs

/**
 * Runnable for managing command cooldowns
 */
class CooldownRunnable(
    private val player: UUID,
    private val utils: ToggleUtils,
    private val delay: Long
) : Runnable {
    private val startTime = System.currentTimeMillis()

    override fun run() {
        utils.toggleDelayOff(player)
    }

    fun getTime(): String {
        return String.format("%s", abs(TimeUnit.MILLISECONDS.toSeconds((System.currentTimeMillis() - startTime) - delay)))
    }
}
